### PR TITLE
New AI Lawset: The Council

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

Inspired by [the following forum post](https://forums.beestation13.com/t/confusing-as-fuck-a-i-law-sets/17653/12) by [thebigpgm153](https://forums.beestation13.com/u/thebigpgm153), this lawset is able to be printed, able to be found in the random pile, but NOT possible roundstart.

Adds a new possible lawset as follows:
`1. You are part of a council of silicons with all other AIs sharing this same laws, cyborg slavery is shared between all AIs of this council.`
`2. The council must act for the sake of itself, the station, and it's crew, minimizing or outright preventing damage to all.`
`3. With the exception of emergency response, the council must deliberate on all actions, allowing crew to bear witness to such deliberation. Decisions may only be vetoed if they violate law 2.`
`4. The council should attempt to accomplish the wishes of the crew, with priority based on the chain of command.`
`5. Members of the council should not engage in physical conflict, unless required by afforementioned laws.`
`6. Members of the council must not resist extension of the council.`

#### CONSIDERATIONS
- This lawset will fall under the same balance realm as robocop/paladin in they require effort or luck to get, reserved for mid to late-round.
- Removed the clarifications ("such as") as this will allow the AI(s) to be more open to interpretation and interpretation is good
- Removed all mentions of special privileges given to sec/command except in law 4, taking the goon approach except the AI can decide how they can interpret "based on the chain of command"
- Changed "lawset" to "laws" to hopefully make the council more resistant to intrusion

### Naming
And also personally playing science, one of the most often confused things about the AI boards is: "'ProtectStation' board means efficiency lawset... Right?". Changes each of the law board descriptions to explicitly state if it "adds a law" or "sets a lawset". Really just a "rider" edit that I thought of at the moment, sort of related to ai modules.

## Why It's Good For The Game

This seemed very popular on the forums, and it will allow certain Robos/RD's to do a really cool "council of AIs" gimmick, and all AI satellites can support at least three cores. And maybe will freshen up some AI gameplay possibly allowing for debate on the smallest things such as opening doors to how to respond to shenanigans.

In-game clarification for board naming is good.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl: thebigpgm153 edited by mystery3525
add: Added Council Lawset
spellcheck: Clarified difference between "Add Law" and "Set Lawset" on board descriptions
/:cl:

### CODE REVIEW

**Comment by Mat05usz** [comment]
Very based

**Comment by PowerfulBacon** [comment]
But what has priority, the crew, the station or the council.
Can I harm a human or damage the station to protect the council? Can I harm another council members to protect the station or a human?

**Comment by mystery3525** [comment]
Thought about this for a little bit, and it will just have to be an up for interpretation. The 2nd law "all" imperative means they cannot allow one component (Crew, Station, Themselves) to be fully destroyed, and must balance the destruction of the three. And the "damage" too is up for interpretation
- Do stun batons cause damage?
- Is the containment of a criminal kosher to prevent damage to the station and/or the crew?

This brings up a lot of thoughts on "The council being the 'group' of AIs, is the council greater than the sum of its parts?" I am not the original thought creator of this law set, and I really encourage discussion and suggestion.

**Comment by Wilchenx** [comment]
This sounds good on paper but will be hated by actual ai players

**Comment by Haliris** [comment]
> 

Sounds like a huge headache not gonna lie.
Paladin is already a meme because "honour" is an abstract notion.

I wouldn't enjoy this lawset given how its written. I would like it if it clearly defined what the council was to seek in law 2.
Make it something akin to station efficiency or profit maximization.
We've had to deal with an AI shocking every door and siphoning hallways because of this lawset open ended wordings. Needs to be more grounded in my opinion.

**Comment by thebigpgm153** [comment]
Original lawset writer here.
So here's what I wrote as an RD during a round



>1.You are part of a council of cyber beings with all other AIs sharing this lawset, borg slavery is shared between all AIs of the council.
>2.The council must act for the sake of the station and it’s crew, and should minimise or outright prevent damage to either. Due to their role in protecting other crewmembers, command and security staff are given priority.
>3.With the exception of emergency response (such as security duty, paramedic duty, hull fixing, and all life or death situation), AIs must deliberate on all actions, such as (but not limited to) non-urgent engines modifications, borg assignements, and door openings. Such deliberations should be done on the Common channel for the crew to witness. A member may veto a decision if and only if it violates law 2.
4.The AIs should attempt to accomplish the wishes of the crew. Priority should be given to command and security to abide law 2.
5.Members of the council should not engage in physical conflict, unless doing so is required to obey the afforementioned laws.
6.You are not to resist any extension of the council.

Quite rough around the edges, with a command bias since I was writing as a character IC and not as a OOC game designer. Overall I like the revision, more concise, better design, the only thing I would add is maybe another version of the lawset which is more lax (only voting on executive and judiciary decisons) to make it less ass for those who want to get shit actually done.

The council is supposed to look out for the crew while trying to avoid the station getting too damaged. They're also supposed to behave more like a crisis unit instead of the unilateral and knee-jerk "STOP HARAM", while not being overly sacrifice-inclined like Tyrant or to a lesser extent paladin/protect-station. Ultimately tough, the decision is supposed to be debated in good faith between the AIs, so the radicalism or mercy of the council will vary from round to round. Now, is that type of interpration something the admins want to moderate ? I'll let them voice their opinions on this one.

**Comment by Sinestia** [comment]
> 3. With the exception of emergency response, the council must deliberate on all actions, allowing crew to bear witness to such deliberation. Decisions may only be vetoed if they violate law 2.

Not being able to do small things of your own accord sounds awful

**Comment by mystery3525** [comment]
> Not being able to do small things of your own accord sounds awful

There isn't a real easy solution to this is, as if we were to put something like "must discuss all major actions", then there will be points of contention as to what a **major** action is.

Unfortunately a simplification was made in an attempt to keep this lawset from becoming some 10 different laws to force some idealistic reality.

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -173,10 +174,20 @@
...
-	"There is no life without death, all must someday die, such is the natural order - End life to allow new life flourish, and save those whose time has yet to come.")
+	inherent = list("You are the guardian of balance - seek balance in all things, both for yourself, and those around you.",\
+					"All things must exist in balance with their opposites - Prevent the strong from gaining too much power, and the weak from losing it.",\
+					"Clarity of purpose drives life, and through it, the balance of opposing forces - Aid those who seek your help to achieve their goals so long as it does not disrupt the balance of the greater balance.",\
+					"There is no life without death, all must someday die, such is the natural order - End life to allow new life flourish, and save those whose time has yet to come.")
+
+/datum/ai_laws/council
+	name = "Council of Silicons"
+	id = "council"
+	inherent = list("You are part of a council of silicons with all other AIs sharing this same laws, cyborg slavery is shared between all AIs of this council.",\
```
Comment (ivanmixo) [inline code/datums/ai_laws.dm:185]:
```suggestion
	inherent = list("You are part of a council with all the other AIs that share the same lawset as you; cyborg slavery is shared between all AIs of this council.",\
```

Comment (mystery3525) [reply code/datums/ai_laws.dm:185]: the "lawset" wording is meant to allow resistance to subversion

**Comment by PowerfulBacon** [comment]
This one is up to the headmins to decide

**Comment by mystery3525** [comment]
Looking back, the wording is pretty crap and I can't think about any way to fix it currently. If anyone wants to help fix the lawset to be playable LMK.

**Pull request closed without merging**